### DISCUSSION TITLE
ch5/ch5-04

### DISCUSSION THREAD

**Discussion by giscus** [gopl-zh/gopl-zh.github.com#224] [General]
# ch5/ch5-04

<The Go Programming Language>中文版

https://golang-china.github.io/gopl-zh/ch5/ch5-04.html

**Comment by soondubu137** [comment]
5.4.0中对于函数出错时如何处理部分返回值的解释与原文有偏差：
> 对于这种情况，正确的处理方式应该是先处理这些不完整的数据，再处理错误。因此对函数的返回值要有清晰的说明，以便于其他人使用。

原文是：
> For correct behavior, some callers may need to process the incomplete data before handling the error, so it is important that such functions clearly document their results.

“为了做出正确的行为，**有些**调用者**或许**需要在处理错误信息前先处理不完整的数据。因此...”

**Comment by soondubu137** [comment]
同样在5.4.0:
> 虽然Go有各种异常机制...

原文是：
> Although Go does have an exception mechanism of sorts...

”虽然**在某种意义上**Go确实有一套异常机制..."

> [!NOTE]
> **of sorts**: of a typical and typically inferior type.
